draft-ietf-moq-transport.md

```diff
@@ -1864,6 +1906,141 @@ SUBSCRIBE_DONE, as defined below:
+  Request ID (i),
+  Track Namespace (tuple),
+  Track Name Length (i),
+  Track Name (..),
+  Group Order (8),
+  Forward (8),
+  ContentExists (8),
+  [Largest (Location),]
+  Number of Parameters (i),
+  Parameters (..) ...,
+}
+~~~
+{: #moq-transport-publish-format title="MOQT PUBLISH Message"}
+
+* Request ID: See {{request-id}}.
+
+* Track Namespace: Identifies a track's namespace as defined in ({{track-name}})
+
+* Track Name: Identifies the track name as defined in ({{track-name}}).
+
+* ContentExists: 1 if an object has been published on this track, 0 if not.
+If 0, then the Largest Group ID and Largest Object ID fields will not be
+present. Any other value is a protocol error and MUST terminate the
+session with a Protocol Violation ({{session-termination}}).
+
+* Largest: The location of the largest object available for this track.
+
+* Forward: The forward mode for this subscription.  Any value other than 0 or 1
+  is a Protocol Violation.  0 indicates the publisher will not transmit any
+  objects until the subscriber sets the Forward State to 1. 1 indicates the
+  publisher will start transmitting objects immediately, even before PUBLISH_OK.
+
+* Parameters: The parameters are defined in {{version-specific-params}}.
+
+
+## PUBLISH_OK {#message-publish-ok}
+
+The subscriber sends an PUBLISH_OK control message to acknowledge the successful
+authorization and acceptance of a PUBLISH message, and establish a subscription.
+
+~~~
+PUBLISH_OK Message
+{
+  Type (i) = 0x7,
+  Length (i),
+  Request ID (i),
+  Subscriber Priority (8),
+  Group Order (8),
+  Forward (8),
+  Filter Type (i),
+  [Start (Location)],
+  [EndGroup (i)],
+  Number of Parameters (i),
+  Parameters (..) ...,
+}
+~~~
+{: #moq-transport-publish-ok format title="MOQT PUBLISH_OK Message"}
+
+* Request ID: The Request ID of the PUBLISH this message is replying to
+  {{message-publish}}.
+
+* Forward: The Forward State for this subscription, either 0 (don't
+  forward) or 1 (forward).
+
+* Subscriber Priority: The Subscriber Priority for this subscription.
+
+* Group Order: The Group Order preference for this subscription.
+
+* Filter Type, Start, End Group: See {{message-subscribe-req}}.
+
+* Parameters: Parameters associated with this message.
+
+## PUBLISH_ERROR {#message-publish-error}
+
+The subscriber sends an PUBLISH_ERROR control message for to reject
+a subscription initiated by PUBLISH.
+
+~~~
+PUBLISH_ERROR Message
+{
+  Type (i) = 0x8,
+  Length (i),
+  Request ID (i),
+  Error Code (i),
+  Reason Phrase Length (i),
+  Reason Phrase (..),
+}
+~~~
+{: #moq-transport-publish-error format title="MOQT PUBLISH_ERROR Message"}
+
+* Request ID: The Request ID of the PUBLISH this message is replying to
+  {{message-publish}}.
+
+* Error Code: Identifies an integer error code for failure.
+
+* Reason Phrase: Provides the reason for error.
+
+The application SHOULD use a relevant error code in PUBLISH_ERROR, as defined
+below:
+
+|------|---------------------------|
```

Comment:
Can we not reuse the SUBSCRIBE error codes?  I think the only different one is "Uninterested"?

Reply:
For now, let's have a separate table, but I'm amendable to consolidating them later.